src/common/datetime.cljs

```diff
@@ -0,0 +1,258 @@
+
+      (within-last-n-days? datetime 1)
+      (str (string/capitalize (i18n/label :t/datetime-yesterday))
+           " "
+           (.format ^js (time-fmt) datetime))
+
+      (within-last-n-days? datetime 6)
+      (.format ^js (datetime-within-one-week-fmt) datetime)
+
+      (current-year? datetime)
+      (.format ^js (short-date-fmt) datetime)
+
+      (previous-years? datetime)
+      (.format ^js (date-fmt) datetime))))
+
+(defn timestamp->mini-date
+  [ms]
+  (.format ^js (short-date-fmt)
+           (-> ms
+               t.coerce/from-long
+               (t/plus time-zone-offset))))
+
+(defn timestamp->time
+  [ms]
+  (.format ^js (time-fmt)
+           (-> ms
+               t.coerce/from-long
+               (t/plus time-zone-offset))))
+
+(defn timestamp->date-key
+  [ms]
+  (keyword (t.format/unparse (t.format/formatter "YYYYMMDD")
+                             (-> ms
+                                 t.coerce/from-long
+                                 (t/plus time-zone-offset)))))
+
+(defn timestamp->long-date
+  [ms]
+  (.format ^js (date-time-fmt)
+           (-> ms
+               t.coerce/from-long
+               (t/plus time-zone-offset))))
+
+(defn format-time-ago
+  [diff unit]
+  (let [name (i18n/label-pluralize diff (:name unit))]
+    (if (= :t/datetime-second-short (:name unit))
+      (i18n/label :t/now)
+      (i18n/label :t/datetime-ago-format-short
+                  {:ago            (i18n/label :t/datetime-ago)
+                   :number         diff
+                   :time-intervals name}))))
+(defn seconds-ago
+  [time]
+  (let [now (t/now)]
+    (if (<= (.getTime ^js time) (.getTime ^js now))
+      (t/in-seconds (t/interval time now))
+      0)))
+
+(defn time-ago
+  [time]
+  (let [diff (seconds-ago time)
+        unit (first (drop-while #(and (>= diff (:limit %))
+                                      (:limit %))
+                                units))]
+    (-> (/ diff (:in-second unit))
+        Math/floor
+        int
+        (format-time-ago unit))))
+
+(defn time-ago-long
+  [time]
+  (let [seconds-ago (seconds-ago time)
+        unit        (first (drop-while #(and (>= seconds-ago (:limit %))
+                                             (:limit %))
+                                       units))
+        diff        (-> (/ seconds-ago (:in-second unit))
+                        Math/floor
+                        int)
+
+        name        (i18n/label-pluralize diff (:name unit))]
+    (i18n/label :t/datetime-ago-format
+                {:ago            (i18n/label :t/datetime-ago)
+                 :number         diff
+                 :time-intervals name})))
+
+(defn to-date
+  [ms]
+  (t.coerce/from-long ms))
+
+(defn timestamp
+  []
+  (inst-ms (js/Date.)))
+
+(defn timestamp-sec
+  []
+  (int (/ (timestamp) 1000)))
+
+(defn timestamp->year-month-day-date
+  [ms]
+  (t.format/unparse (:year-month-day t.format/formatters) (to-date ms)))
+
+(re-frame/reg-cofx
```

Comment:
this shouldn't be here for sure , somewhere in setup events

Reply:
cool, I can move that too

Reply:
was it moved ?

Reply:
I moved it now to status-im2.setup.datetime